index.js

```diff
@@ -25,10 +25,11 @@ var KNOWN_STATICS = {
 };
 
 module.exports = function hoistNonReactStatics(targetComponent, sourceComponent) {
-    var keys = Object.getOwnPropertyNames(sourceComponent);
-    for (var i=0; i<keys.length; ++i) {
-        if (!REACT_STATICS[keys[i]] && !KNOWN_STATICS[keys[i]]) {
-            targetComponent[keys[i]] = sourceComponent[keys[i]];
+    var sourceKeys = Object.getOwnPropertyNames(sourceComponent),
+        targetKeys = Object.getOwnPropertyNames(targetComponent);
+    for (var i=0; i<sourceKeys.length; ++i) {
+        if (!REACT_STATICS[sourceKeys[i]] && !KNOWN_STATICS[sourceKeys[i]] && targetKeys.indexOf(sourceKeys[i]) === -1) {
```

Comment:
Behavior shouldn't be different, but I wonder if it's inheriting that property instead of it being declared directly on the object. That code is pretty confusing, and I'm not sure which component is which.

You mentioned that the code you have here wasn't working either though and that you might need try/catch. Would it be more efficient to *just* do try/catch for now and remove the check?